src/hooks/sync-sites/sync-sites-context.tsx

```diff
@@ -1,24 +1,67 @@
-import React, { createContext, useContext, useState } from 'react';
+import { __, sprintf } from '@wordpress/i18n';
+import React, { createContext, useCallback, useContext, useState } from 'react';
+import { getIpcApi } from '../../lib/get-ipc-api';
 import { SyncSite } from '../use-fetch-wpcom-sites';
-import { usePullPushTimestamps } from '../use-pull-push-timestamps';
+import { useFormatLocalizedTimestamps } from '../use-format-localized-timestamps';
 import { useListenDeepLinkConnection } from './use-listen-deep-link-connection';
 import { useSiteSyncManagement } from './use-site-sync-management';
 import { useSyncPull } from './use-sync-pull';
 import { useSyncPush } from './use-sync-push';
 
 export type SyncSitesContextType = ReturnType< typeof useSyncPull > &
 	ReturnType< typeof useSyncPush > &
-	ReturnType< typeof useSiteSyncManagement > &
-	ReturnType< typeof usePullPushTimestamps >;
+	ReturnType< typeof useSiteSyncManagement > & {
+		getLastSyncTimeText: ( timestamp: string | null, type: 'pull' | 'push' ) => string;
+		updateSiteTimestamp: ( site: SyncSite, type: 'pull' | 'push' ) => Promise< void >;
+	};
 
 const SyncSitesContext = createContext< SyncSitesContextType | undefined >( undefined );
 
 export function SyncSitesProvider( { children }: { children: React.ReactNode } ) {
+	const { formatRelativeTime } = useFormatLocalizedTimestamps();
 	const [ pullStates, setPullStates ] = useState< SyncSitesContextType[ 'pullStates' ] >( {} );
+	const [ connectedSites, setConnectedSites ] = useState< SyncSite[] >( [] );
+
+	const getLastSyncTimeText = useCallback(
+		( timestamp: string | null, type: 'pull' | 'push' ): string => {
+			if ( timestamp === null ) {
+				return type === 'pull'
+					? __( 'You have not pulled this site yet.' )
+					: __( 'You have not pushed this site yet.' );
+			}
+
+			return sprintf(
+				type === 'pull'
+					? __( 'You pulled this site %s ago.' )
+					: __( 'You pushed this site %s ago.' ),
+				formatRelativeTime( timestamp )
+			);
+		},
+		[ formatRelativeTime ]
+	);
+
+	const updateSiteTimestamp = useCallback( async ( site: SyncSite, type: 'pull' | 'push' ) => {
+		try {
+			const updatedSite = {
+				...site,
+				[ type === 'pull' ? 'lastPullTimestamp' : 'lastPushTimestamp' ]: new Date().toISOString(),
+			};
+
+			await getIpcApi().updateSingleConnectedWpcomSite( updatedSite );
+			setConnectedSites( ( sites ) =>
+				sites.map( ( s ) => ( s.id === site.id ? updatedSite : s ) )
+			);
```

Comment:
Here's an idea that could potentially work

```ts

	const [ connectedSiteToUpdate, setConnectedSiteToUpdate ] =
		useState< [ SyncSite[ 'id' ], 'push' | 'pull' ] >();

	const { pullSite, isAnySitePulling, isSiteIdPulling, clearPullState, getPullState } = useSyncPull(
		{
			pullStates,
			setPullStates,
			onPullSuccess: ( siteId ) => setConnectedSiteToUpdate( [ siteId, 'pull' ] ),
			connectedSites,
		}
	);

	const [ pushStates, setPushStates ] = useState< SyncSitesContextType[ 'pushStates' ] >( {} );
	const { pushSite, isAnySitePushing, isSiteIdPushing, clearPushState, getPushState } = useSyncPush(
		{
			pushStates,
			setPushStates,
			onPushSuccess: ( siteId ) => setConnectedSiteToUpdate( [ siteId, 'push' ] ),
			connectedSites,
		}
	);

	useEffect( () => {
		if ( ! connectedSiteToUpdate ) {
			return;
		}

		const updatedConnectedSites = connectedSites.map( ( s ) => {
			if ( s.id === connectedSiteToUpdate[ 0 ] ) {
				return {
					...s,
					[ connectedSiteToUpdate[ 1 ] === 'pull' ? 'lastPullTimestamp' : 'lastPushTimestamp' ]:
						new Date().toISOString(),
				};
			}

			return s;
		} );

		setConnectedSites( updatedConnectedSites );
		getIpcApi().updateConnectedWpcomSites( updatedConnectedSites );
	}, [ connectedSiteToUpdate ] );
```

Reply:
I don't want to overcomplicate the review of this PR, but with the changes in https://github.com/Automattic/studio/commit/40724600c3658b59c8a39e6b72aba8777c271246, we are even closer to implementing a solution like this ☝️

The benefit being that we can remove `updateSingleConnectedWpcomSite` from `ipc-handlers` if we update the timestamp in a `useEffect` that reacts to a state change instead of in a memoized callback (like we do now).

Reply:
Thanks for the suggestion, I like that we can potentially re-use what is already in there! I tested this approach briefly and I think it is something that could be done but there are some details that would need to be taken care of.  I also think it is okay to have that handler in the `ipc-handlers` for now. It might become handy when we might need to update a specific connected site for some other reason (we also have a similar function for[ updating a local site](https://github.com/Automattic/studio/blob/a5392f7bffafddcbc3c1a94b4ae9f53cd96ff40a/src/ipc-handlers.ts#L206).

Let's create a follow-up issue where we can consider this refactor in detail. I suggest that we proceed with merging as is for now considering the timing (we are about to launch the sync feature) and the urgency (we still have lots of issues to get through) and handle this as a follow-up issue. I will open a new issue for this.

Reply:
Created follow up here: https://github.com/Automattic/dotcom-forge/issues/10085

Reply:
> I don't want to overcomplicate the review of this PR, but with the changes in https://github.com/Automattic/studio/commit/40724600c3658b59c8a39e6b72aba8777c271246, we are even closer to implementing a solution like this ☝️

Oh, I just saw the comment but I already saw the changes. I will handle it in the follow up issue that I opened 👍